keel-web/src/main/kotlin/com/netflix/spinnaker/config/DefaultConfiguration.kt

```diff
@@ -61,13 +60,11 @@ class DefaultConfiguration(
   @Bean
   fun idGenerator(): ULID = ULID()
 
-  @Bean
+  @Bean(name = ["jsonMapper", "objectMapper"])
```

Comment:
Using a `@Qualifier` instead of bean names was causing issues with auto-detection of auto-wiring in IntelliJ.

Reply:
So, now we have 2 beans? The same bean registered twice? What happens if something injects `Collection<ObjectMapper>` (e.g. where we register sub-types), does the collection have 2 elements or 3? I think the point of naming it `jsonMapper` was to distinguish from `yamlMapper` which is also an instance of `ObjectMapper`.

Reply:
That's still a problem, actually, because they're both the same type. This is not registering two beans, just having a primary name and an alias for the same bean. For consumers that only have 1 `ObjectMapper` injected, if it's named either of those names, auto-wiring will work automatically.

From [the docs](https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/context/annotation/Bean.html):
>**Bean Names**
> While a `name()` attribute is available, the default strategy for determining the name of a bean is to use the name of the `@Bean` method. This is convenient and intuitive, but if explicit naming is desired, the name attribute (or its alias value) may be used. Also note that name accepts an array of Strings, allowing for multiple names (i.e. a primary bean name plus one or more aliases) for a single bean.

Reply:
This is not registering two beans, just having a primary name and an alias for the same bean. This now allows for auto-wiring with either of those names, without the need to use a `@Qualifier` at the point of injection. Auto-wiring for `yamlMapper` also continues to work.

From [the docs](https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/context/annotation/Bean.html):
>**Bean Names**
> While a `name()` attribute is available, the default strategy for determining the name of a bean is to use the name of the `@Bean` method. This is convenient and intuitive, but if explicit naming is desired, the name attribute (or its alias value) may be used. Also note that name accepts an array of Strings, allowing for multiple names (i.e. a primary bean name plus one or more aliases) for a single bean.